### PR TITLE
debug: enable profiling, print to stderr 

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

**Review by jorblancoa — APPROVED** [review]

File: var/spack/repos/builtin/packages/synapsetool/package.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ class Synapsetool(CMakePackage):
     git      = "ssh://bbpcode.epfl.ch/hpc/synapse-tool"
 
     version('develop', submodules=True)
+    version('0.5.4', tag='v0.5.3', submodules=True)
```
Comment (jorblancoa) [inline var/spack/repos/builtin/packages/synapsetool/package.py:39]: typo on tag?

Comment (matz-e) [reply var/spack/repos/builtin/packages/synapsetool/package.py:39]: No, that's intended. This is WIP, triggering a rebuild for debugging purpose.

**Comment by ferdonline** [comment]
This is not supposed to be merged, right?

**Comment by matz-e** [comment]
> This is not supposed to be merged, right?

That's why I marked it "work in progress"…

**Comment by matz-e** [comment]
Note to self:
```[2019-09-24T07:03:02.245Z] ### 09:03:01 gathering specs
[2019-09-24T07:08:39.384Z]          488645763 function calls (479041717 primitive calls) in 334.845 seconds
[2019-09-24T07:08:39.385Z] 
[2019-09-24T07:08:39.385Z]    Ordered by: cumulative time
[2019-09-24T07:08:39.385Z]    List reduced from 7582 to 20 due to restriction <20>
[2019-09-24T07:08:39.385Z] 
[2019-09-24T07:08:39.385Z]    ncalls  tottime  percall  cumtime  percall filename:lineno(function)
[2019-09-24T07:08:39.385Z]         1    0.000    0.000  334.876  334.876 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/main.py:433(_invoke_command)
[2019-09-24T07:08:39.385Z]         1    0.001    0.001  334.876  334.876 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/cmd/filter.py:71(filter)
[2019-09-24T07:08:39.385Z]        50    0.000    0.000  325.099    6.502 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1978(concretized)
[2019-09-24T07:08:39.385Z]        50    0.036    0.001  325.091    6.502 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1841(concretize)
[2019-09-24T07:08:39.385Z]       173    0.189    0.001  259.781    1.502 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1737(_expand_virtual_packages)
[2019-09-24T07:08:39.385Z]       757    0.008    0.000  148.221    0.196 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:2237(normalize)
[2019-09-24T07:08:39.385Z]     12307    0.055    0.000  146.988    0.012 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/concretize.py:111(choose_virtual_or_external)
[2019-09-24T07:08:39.385Z]     12307    0.976    0.000  144.532    0.012 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/concretize.py:59(_valid_virtuals_and_externals)
[2019-09-24T07:08:39.385Z]     25763    2.621    0.000  141.826    0.006 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/package_prefs.py:28(get_packages_config)
[2019-09-24T07:08:39.385Z]     25790    0.051    0.000  115.742    0.004 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/config.py:666(get)
[2019-09-24T07:08:39.385Z]     25790    0.057    0.000  115.663    0.004 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/config.py:504(get)
[2019-09-24T07:08:39.385Z]     25790    0.304    0.000  115.597    0.004 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/config.py:451(get_config)
[2019-09-24T07:08:39.385Z] 361069/51601    6.587    0.000  114.397    0.002 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/config.py:796(_merge_yaml)
[2019-09-24T07:08:39.385Z] 90910/757    0.529    0.000  105.811    0.140 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:2200(_normalize_helper)
[2019-09-24T07:08:39.385Z] 90153/8896    2.495    0.000  100.692    0.011 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:2100(_merge_dependency)
[2019-09-24T07:08:39.385Z]   2344996    4.046    0.000   93.638    0.000 /usr/lib64/python2.7/copy.py:66(copy)
[2019-09-24T07:08:39.385Z]     12881    0.019    0.000   79.241    0.006 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/package_prefs.py:225(is_spec_buildable)
[2019-09-24T07:08:39.385Z]     12881    0.174    0.000   63.472    0.005 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/package_prefs.py:190(spec_externals)
[2019-09-24T07:08:39.385Z]   4681965   15.368    0.000   61.163    0.000 /usr/lib64/python2.7/collections.py:38(__init__)
[2019-09-24T07:08:39.385Z]   2319146    5.898    0.000   47.050    0.000 /usr/lib64/python2.7/copy.py:306(_reconstruct)
```
```
[2019-09-24T07:16:48.924Z] [+] /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/deploy/applications/2019-09-23/linux-rhel7-x86_64/gcc-6.4.0/parquet-converters-0.5.1-btljqpwgk2
[2019-09-24T07:21:56.004Z]          428533607 function calls (390174620 primitive calls) in 297.947 seconds
[2019-09-24T07:21:56.004Z] 
[2019-09-24T07:21:56.004Z]    Ordered by: cumulative time
[2019-09-24T07:21:56.004Z]    List reduced from 2016 to 20 due to restriction <20>
[2019-09-24T07:21:56.004Z] 
[2019-09-24T07:21:56.004Z]    ncalls  tottime  percall  cumtime  percall filename:lineno(function)
[2019-09-24T07:21:56.004Z]         1    0.000    0.000  298.231  298.231 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/main.py:433(_invoke_command)
[2019-09-24T07:21:56.004Z]         1    0.000    0.000  298.231  298.231 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/cmd/module.py:56(module)
[2019-09-24T07:21:56.004Z]         1    0.000    0.000  298.231  298.231 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/cmd/modules/__init__.py:394(modules_cmd)
[2019-09-24T07:21:56.004Z]         1    0.004    0.004  284.109  284.109 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/cmd/modules/__init__.py:282(refresh)
[2019-09-24T07:21:56.005Z]        63    0.014    0.000  213.603    3.391 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:694(write)
[2019-09-24T07:21:56.005Z] 5803683/2604468    1.217    0.000  186.550    0.000 {getattr}
[2019-09-24T07:21:56.005Z]        63    0.001    0.000  186.022    2.953 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/tengine.py:65(to_dict)
[2019-09-24T07:21:56.005Z]        63    0.000    0.000  126.477    2.008 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:632(autoload)
[2019-09-24T07:21:56.005Z]       126    0.001    0.000  126.477    1.004 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:641(_create_module_list_of)
[2019-09-24T07:21:56.005Z]        63    0.001    0.000  121.863    1.934 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:405(specs_to_load)
[2019-09-24T07:21:56.005Z]       126    0.022    0.000  121.863    0.967 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:425(_create_list_for)
[2019-09-24T07:21:56.005Z]      2262    0.082    0.000  107.895    0.048 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:359(blacklisted)
[2019-09-24T07:21:56.005Z]      5334    0.331    0.000  105.715    0.020 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:663(_read)
[2019-09-24T07:21:56.005Z]      5334    0.467    0.000  105.275    0.020 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:391(_read_from_file)
[2019-09-24T07:21:56.005Z]      5329    0.016    0.000  104.988    0.020 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/llnl/util/lock.py:376(__enter__)
[2019-09-24T07:21:56.005Z] 793542/22808    0.729    0.000  104.360    0.005 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:348(query_by_spec_hash)
[2019-09-24T07:21:56.005Z]      2267    0.005    0.000   89.397    0.039 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:72(converter)
[2019-09-24T07:21:56.005Z]      2267    0.005    0.000   89.392    0.039 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:816(get_record)
[2019-09-24T07:21:56.005Z]      2262    0.010    0.000   89.228    0.039 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:3226(_installed_explicitly)
[2019-09-24T07:21:56.005Z]       706    0.008    0.000   76.646    0.109 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:465(use_name)
[2019-09-24T07:21:56.005Z] 
[2019-09-24T07:21:56.005Z] 
[2019-09-24T07:21:56.005Z] ==> Regenerating tcl module files
```

**Comment by matz-e** [comment]
Caching:
```
[2019-09-24T09:49:37.798Z] ### 11:49:36 gathering specs
[2019-09-24T09:53:14.733Z]          304806062 function calls (295231267 primitive calls) in 212.358 seconds
[2019-09-24T09:53:14.733Z] 
[2019-09-24T09:53:14.733Z]    Ordered by: cumulative time
[2019-09-24T09:53:14.733Z]    List reduced from 7582 to 20 due to restriction <20>
[2019-09-24T09:53:14.733Z] 
[2019-09-24T09:53:14.733Z]    ncalls  tottime  percall  cumtime  percall filename:lineno(function)
[2019-09-24T09:53:14.733Z]         1    0.000    0.000  212.389  212.389 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/main.py:433(_invoke_command)
[2019-09-24T09:53:14.733Z]         1    0.001    0.001  212.389  212.389 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/cmd/filter.py:71(filter)
[2019-09-24T09:53:14.733Z]        50    0.000    0.000  200.630    4.013 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1978(concretized)
[2019-09-24T09:53:14.733Z]        50    0.035    0.001  200.621    4.012 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1841(concretize)
[2019-09-24T09:53:14.733Z]       757    0.008    0.000  147.496    0.195 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:2237(normalize)
[2019-09-24T09:53:14.733Z]       173    0.089    0.001  135.656    0.784 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1737(_expand_virtual_packages)
[2019-09-24T09:53:14.733Z] 90910/757    0.523    0.000  104.681    0.138 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:2200(_normalize_helper)
[2019-09-24T09:53:14.733Z] 90153/8896    2.495    0.000   99.893    0.011 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:2100(_merge_dependency)
[2019-09-24T09:53:14.733Z]    116323    0.614    0.000   45.310    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:2033(_evaluate_dependency_conditions)
[2019-09-24T09:53:14.733Z]       757    0.267    0.000   38.722    0.051 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:2307(validate_or_raise)
[2019-09-24T09:53:14.733Z]    707544    1.288    0.000   37.510    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1208(package_class)
[2019-09-24T09:53:14.733Z]   5434532    2.315    0.000   37.012    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1215(virtual)
[2019-09-24T09:53:14.733Z]    707544    0.587    0.000   35.177    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/repo.py:552(get_pkg_class)
[2019-09-24T09:53:14.733Z]    279703    1.383    0.000   34.769    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1044(__init__)
[2019-09-24T09:53:14.733Z]   5434534    7.027    0.000   34.696    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1226(is_virtual)
[2019-09-24T09:53:14.733Z] 121087/40893    0.189    0.000   34.393    0.001 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/repo.py:79(converter)
[2019-09-24T09:53:14.733Z] 57360/37661    0.105    0.000   33.695    0.001 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/repo.py:1103(get)
[2019-09-24T09:53:14.733Z]    764904    1.499    0.000   32.707    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/repo.py:984(get_pkg_class)
[2019-09-24T09:53:14.733Z]    279814    0.307    0.000   32.019    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/parse.py:150(parse)
[2019-09-24T09:53:14.733Z] 57360/37661    0.068    0.000   29.345    0.001 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/repo.py:544(get)
```
```
[2019-09-24T10:01:22.582Z] ==> Successfully installed parquet-converters
[2019-09-24T10:01:22.582Z]   Fetch: 20.37s.  Build: 54.51s.  Total: 1m 14.88s.
[2019-09-24T10:01:22.582Z] [+] /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/deploy/applications/2019-09-23/linux-rhel7-x86_64/gcc-6.4.0/parquet-converters-0.5.1-eqncjyc526
[2019-09-24T10:07:29.752Z]          516315944 function calls (468136553 primitive calls) in 357.665 seconds
[2019-09-24T10:07:29.752Z] 
[2019-09-24T10:07:29.752Z]    Ordered by: cumulative time
[2019-09-24T10:07:29.752Z]    List reduced from 2016 to 20 due to restriction <20>
[2019-09-24T10:07:29.752Z] 
[2019-09-24T10:07:29.752Z]    ncalls  tottime  percall  cumtime  percall filename:lineno(function)
[2019-09-24T10:07:29.752Z]         1    0.000    0.000  357.952  357.952 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/main.py:433(_invoke_command)
[2019-09-24T10:07:29.752Z]         1    0.000    0.000  357.952  357.952 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/cmd/module.py:56(module)
[2019-09-24T10:07:29.752Z]         1    0.000    0.000  357.952  357.952 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/cmd/modules/__init__.py:394(modules_cmd)
[2019-09-24T10:07:29.752Z]         1    0.005    0.005  343.138  343.138 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/cmd/modules/__init__.py:282(refresh)
[2019-09-24T10:07:29.752Z]        63    0.016    0.000  263.729    4.186 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:694(write)
[2019-09-24T10:07:29.752Z] 5873322/2629505    1.245    0.000  234.849    0.000 {getattr}
[2019-09-24T10:07:29.752Z]        63    0.001    0.000  234.307    3.719 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/tengine.py:65(to_dict)
[2019-09-24T10:07:29.752Z]        63    0.000    0.000  168.517    2.675 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:632(autoload)
[2019-09-24T10:07:29.752Z]       126    0.001    0.000  168.517    1.337 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:641(_create_module_list_of)
[2019-09-24T10:07:29.752Z]        63    0.001    0.000  163.855    2.601 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:405(specs_to_load)
[2019-09-24T10:07:29.752Z]       126    0.022    0.000  163.855    1.300 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:425(_create_list_for)
[2019-09-24T10:07:29.752Z]      5358    0.497    0.000  163.518    0.031 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:663(_read)
[2019-09-24T10:07:29.752Z]      5358    0.634    0.000  162.909    0.030 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:391(_read_from_file)
[2019-09-24T10:07:29.752Z]      5353    0.017    0.000  162.825    0.030 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/llnl/util/lock.py:376(__enter__)
[2019-09-24T10:07:29.752Z] 1187839/26482    1.255    0.000  161.500    0.006 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:348(query_by_spec_hash)
[2019-09-24T10:07:29.752Z]      2270    0.083    0.000  157.168    0.069 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/modules/common.py:359(blacklisted)
[2019-09-24T10:07:29.752Z]      2275    0.005    0.000  138.939    0.061 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:72(converter)
[2019-09-24T10:07:29.752Z]      2275    0.006    0.000  138.934    0.061 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/database.py:816(get_record)
[2019-09-24T10:07:29.752Z]      2270    0.011    0.000  138.677    0.061 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:3226(_installed_explicitly)
[2019-09-24T10:07:29.752Z]   4594616    5.167    0.000   85.583    0.000 /gpfs/bbp.cscs.ch/apps/hpc/jenkins/pulls/523/spack/lib/spack/spack/spec.py:1245(traverse)
```

**Pull request closed without merging**